cmd/thanos/compact.go

```diff
@@ -349,6 +349,7 @@ func runCompact(
 		reg,
 		compactMetrics.blocksMarked.WithLabelValues(metadata.DeletionMarkFilename),
 		compactMetrics.garbageCollectedBlocks,
+		compactMetrics.blocksMarked.WithLabelValues(metadata.NoCompactMarkFilename),
```

Comment:
I am wondering if we need a new `reason` label for this mark counter. Now we cannot differentiate any compact blocks due to large index or out of order chunks.

Reply:
sounds reasonable, especially if we want to skip all blocks with critical error as @alvinlin123 suggested